### PR TITLE
Add uid validation comments

### PR DESCRIPTION
**Pull request by Aradon267**

### CODE REVIEW

File: cv_next/app/actions/comments/deleteComment.ts [hunk 1]
```diff
@@ -13,6 +14,32 @@ import { logErrorWithTrace } from "@/server/base/logger";
...
+    currentIdResult.val != commentResult.val.user_id &&
+    !resultAdminCheck.ok
+  ) {
+    logger.error(
+      `${currentIdResult.val} tried deleting someone elses comment!`
+    );
+    return Err(
+      "An error has occurred while deleting the comment. Please try again later."
+    );
+  }
```
Comment (YechiamW3) [inline cv_next/app/actions/comments/deleteComment.ts:42]: Please move the validation checks to a separate function (in this file),
such as you'd have:
```
const commentResult = await getCommentById(commentId);
const validationError = validateUserAction(...);
if (validationError) return validationError;
```

Nicely encapsulating the mess of validation :)

File: cv_next/server/api/comments.ts [hunk 2]
```diff
@@ -28,12 +28,13 @@ export async function addCommentToCv(
...
         addCommentToCv.name,
         undefined,
+        undefined,
         new Error("adding comment failed")
       );
     }
     return Ok.EMPTY;
   } catch (err) {
-    return Err(addCommentToCv.name, undefined, err as Error);
+    return Err(addCommentToCv.name, undefined, undefined, err as Error);
```
Comment (YechiamW3) [inline cv_next/server/api/comments.ts:37]: Instead perhaps define Err with an object of errors (which can then be either `PostgresError`, `AuthError` etc)?

File: cv_next/app/actions/comments/deleteComment.ts (hunk 1 above)
Comment (Aradon267) [reply cv_next/app/actions/comments/deleteComment.ts:42]: Did it only to this file, can do to others if needed

Comment (YechiamW3) [reply cv_next/app/actions/comments/deleteComment.ts]: Wherever you think is needed, in this file it's particularly jarring

**Review by YechiamW3 — APPROVED** [review]

**Pull request merged**